multidict/_multidict_py.py

```diff
@@ -530,6 +568,14 @@ def _replace(self, key: str, value: _V) -> None:
 class CIMultiDict(MultiDict[_V]):
     """Dictionary with the support for duplicate case-insensitive keys."""
 
+    def _key(self, key: str) -> str:
+        if isinstance(key, str):
+            if getattr(type(key), '__is_istr__', False):
+                return key
+            else:
+                return istr(key)
```

Comment:
I'm not clear I understand this part? Is there expected to be some other custom types implemented by users?

Seems to me like if the condition is true, then the type is probably istr, in which case it's be the same to do:
```suggestion
            return key
```